src/blenderbim/blenderbim/bim/module/material/operator.py

```diff
@@ -621,18 +621,12 @@ def load_profile_attributes(self, material_set_item, material_set_item_data):
                 new.is_null = profile_data[attribute.name()] is None
                 new.is_optional = attribute.optional()
                 new.data_type = data_type
-                if data_type == "string":
-                    new.string_value = "" if new.is_null else profile_data[attribute.name()]
-                elif data_type == "float":
-                    new.float_value = 0.0 if new.is_null else profile_data[attribute.name()]
-                elif data_type == "integer":
-                    new.int_value = 0 if new.is_null else profile_data[attribute.name()]
-                elif data_type == "boolean":
-                    new.bool_value = False if new.is_null else profile_data[attribute.name()]
-                elif data_type == "enum":
+                if data_type == "enum":
                     new.enum_items = json.dumps(ifcopenshell.util.attribute.get_enum_items(attribute))
                     if profile_data[attribute.name()]:
                         new.enum_value = profile_data[attribute.name()]
+                else:
+                    new.set_value(new.get_value_default() if new.is_null else profile_data[attribute.name()])
```

Comment:
`import_attributes` helper, but check that it doesn't break anything. May require a callback. May not.

Reply:
In the end I prefer not risking breaking things I don't have the data to test against :)